### PR TITLE
ADBDEV-3099: Add support year if it contains more than 4 digits

### PR DESCRIPTION
**Pull request by RomaZe**

### CODE REVIEW

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java [hunk 1]
```diff
@@ -311,19 +339,50 @@ public static void decodeOneRowToPreparedStatement(OneRow row, PreparedStatement
...
                 default:
                     throw new IOException("The data tuple from JdbcResolver is corrupted");
             }
         }
     }
+
+    private Object getLocalDate(String rawVal) {
+        try {
+            String yearStr = rawVal.trim().substring(0, rawVal.indexOf("-"));
+            return yearStr.length() > 4 ? LocalDate.parse("+" + rawVal) : LocalDate.parse(rawVal);
```
Comment (xardazzzzzz) [inline server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java]: What does this appended "+" sign means?

Comment (RomaZe) [reply server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java]: https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_LOCAL_DATE
Four digits or more for the year. Years in the range 0000 to 9999 will be pre-padded by zero to ensure four digits. Years outside that range will have a prefixed positive or negative symbol.

So, date and timestamp values come from the table as a `String`. The string doesn't contain any prefix if the year contains more than 4 digits. But to convert the string to LocalDate with `parse` method we have to add positive or negative symbol.

But it is a good point as I realized that dates might start with `-`, which means BC. I will try do deal with this.

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java [hunk 2]
```diff
@@ -36,15 +36,59 @@
...
+            .appendValue(ChronoField.MONTH_OF_YEAR, 2).appendLiteral('-')
+            .appendValue(ChronoField.DAY_OF_MONTH, 2)
+            .optionalStart().appendPattern(" G").optionalEnd()
+            .toFormatter();
+
+    private static final DateTimeFormatter LOCAL_DATE_TIME_SET_FORMATTER = (new DateTimeFormatterBuilder())
+            .appendValue(ChronoField.YEAR_OF_ERA, 1, 9, SignStyle.NORMAL).appendLiteral('-')
+            .appendValue(ChronoField.MONTH_OF_YEAR, 1, 2, SignStyle.NORMAL).appendLiteral('-')
+            .appendValue(ChronoField.DAY_OF_MONTH, 1, 2, SignStyle.NORMAL)
+            .appendLiteral(" ")
```
Comment (xardazzzzzz) [inline server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java:87]: Why do we use space as separator and not T?

Comment (RomaZe) [reply server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java:87]: Due to Date\Time output format from the documentation: https://www.postgresql.org/docs/9.4/datatype-datetime.html#DATATYPE-DATETIME-OUTPUT

I also should notice that the PXF JDBC supports only `ISO 8601, SQL standard` date style (1997-12-17 07:37:16-08). It doesn't connect with this PR. I guess, the reason is that Greeenplum sends `Date` and `Timestamp` to PXF as a `String` and not as actual Date or Timestamp. So we should parse String and convert it to LocalDate or LocalDateTime.

**Review by xardazzzzzz — APPROVED** [review]

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java (hunk 1 above)
Comment (RomaZe) [reply server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java]: Added support date with BC. I replaced all string replacing with DateTimeFormatter class. I think it's much better.

**Review by iamlapa — APPROVED** [review]

**Pull request merged**